src/BodyDigest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace HttpSignatures;
+
+class BodyDigest
+{
+    /** @var string */
+    private $hashName;
+
+    /** @var string */
+    private $digestHeaderPrefix;
+
+    /**
+     * @param string $name
+     * @return BodyDigest
+     * @throws DigestException
+     */
+    public function __construct($hashAlgorithm = null)
+    {
+        switch (strtolower(str_replace("-", '', $hashAlgorithm))) {
+        case 'sha':
+        case 'sha1':
+            $this->hashName = 'sha1';
+            $this->digestHeaderPrefix = 'SHA';
+            break;
+        case 'sha256':
+        case null:
+        case '':
+            $this->hashName = 'sha256';
+            $this->digestHeaderPrefix = 'SHA-256';
+            break;
+        case 'sha512':
+            $this->hashName = 'sha512';
+            $this->digestHeaderPrefix = 'SHA-512';
+            break;
+        default:
+            throw new DigestException("Digest algorithm parameter '$hashAlgorithm' not understood");
+            break;
+        }
+    }
```

Comment:
This function looks more like a factory function.. e.g `public static function create($hashAlgorithm)`, while the constructor should take `$hashName` and `$digestHeaderPrefix`

Reply:
This constructor is tolerant of variations: ``SHA256``, ``SHA-256`` and ``sha256`` are all equivalent and valid inputs for the same hash function, but generate the right openssl name and Digest name without expecting the user to know exactly how each is formatted.

Rather let the user be lazy and we fix it up, than expect the user to supply an exactly correct string and matching values for the two parameters which we would have to sanity check.

It also has a sane default if not value is provided.

Reply:
@mtibben Is this acceptable?